src/aggregationserver/databaseclient.cpp

```diff
@@ -466,6 +466,52 @@ const pqxx::result DatabaseClient::GetMemUtilInfo(
     }
 }
 
+void DatabaseClient::UpdateShutdownTime(
+    int experiment_run_id,
+    std::string end_time
+) {
+    std::stringstream query_stream;
+
+    query_stream << "UPDATE ExperimentRun SET EndTime = " << connection_.quote(end_time);
+    query_stream << "WHERE ExperimentRunID = " << experiment_run_id;
+
+    std::lock_guard<std::mutex> conn_guard(conn_mutex_);
+
+    try {
+        pqxx::work transaction(connection_, "UpdateShutdownTransaction");
+        const auto& pg_result = transaction.exec(query_stream.str());
+        transaction.commit();
+    } catch (const std::exception& e)  {
+        std::cerr << "An exception occurred while updating experiment run shutdown time: " << e.what() << std::endl;
+        throw;
+    }
+}
+
+void DatabaseClient::UpdateLastSampleTime(
+    int experiment_run_id,
+    std::string sample_time
+) {
+    std::stringstream query_stream;
+
+    std::string timeval = connection_.quote(sample_time);
```

Comment:
Might be a good idea to rename this: http://pubs.opengroup.org/onlinepubs/7908799/xsh/systime.h.html

Reply:
time_val also fits our naming convention more correctly.

Reply:
Addresed in [this commit](https://github.com/cdit-ma/logan/pull/40/commits/1334d762696ed12fafe4e5cf65bbba7f98218b84)